doc/man/config.rst

```diff
@@ -0,0 +1,49 @@
+Config file support
+===================
+
+Config files are supported by most subcommands of ``pkgdev`` from any of three
+locations. Listed in order of increasing precedence these include the
+following:
+
+- system config -- /etc/pkgdev/pkgdev.conf
+- user config -- ~/.config/pkgdev/pkgdev.conf
+- custom config -- specified via the --config option
+
+Any settings from a config file with higher precedence will override matching
+settings from a config file with a lower precedence, e.g. user settings
+override system settings. Note that command line options override any matching
+config file setting.
+
+In terms of file structure, basic INI formatting is required and allows
+creating a default section (DEFAULT) for system-wide settings or repo-specific
+sections. The INI key-value pairs directly relate to the available
+long-options supported by the various prefixed by the subcommand name and their
+related values. To find all possible configuration options, run:
+``pkgdev {subcommand} --help``. See the following examples for config settings:
```

Comment:
I'm not sure if I get this. If I run `pkgdev commit --help`, I get:
```
usage: pkgdev commit [--config CONFIG_FILE] [-h] [--debug] [-q] [-v] [--color BOOLEAN] [-b BUG] [-c CLOSES] [-T NAME:VALUE] [-n] [-s [BOOLEAN]] [-A] [--mangle [BOOLEAN]]
                     [-m MSG | -M FILE] [-u | -a]

create git commit

config options:
  --config CONFIG_FILE  use custom pkgdev settings file

base options:
  -h, --help            show this help message and exit
  --debug               enable debugging checks
  -q, --quiet           suppress non-error messages
  -v, --verbose         show verbose output
  --color BOOLEAN       enable/disable color support

commit options:
  -b BUG, --bug BUG     add Bug tag for a given Gentoo or upstream bug
  -c CLOSES, --closes CLOSES
                        add Closes tag for a given Gentoo bug or upstream PR URL
  -T NAME:VALUE, --tag NAME:VALUE
                        add commit tag
  -n, --dry-run         pretend to create commit
  -s [BOOLEAN], --scan [BOOLEAN]
                        run pkgcheck against staged changes
  -A, --ask             confirm creating commit with QA errors
  --mangle [BOOLEAN]    forcibly enable/disable file mangling
  -m MSG, --message MSG
                        specify commit message
  -M FILE, --message-template FILE
                        use commit message template from specified file
  -u, --update          stage all changed files
  -a, --all             stage all changed/new/removed files
```

How does this tell me what I can set in a config file?

Reply:
`--help` output != man page content

It's done correct here, use them man page for long, structured, custom content.

Reply:
OK, I can see it might be not well understood the working of long option into config.

An example:
Let's say you like to always use `pkgdev commit -as` (looking at myself), and you want to auto add those two. You look at help and see this is equal to long option: `pkgdev commit --ask --scan true` (at least as stated by above help). Now you set it as: `--ask` without value, and `--scan` with value `true`, so you make it as:
```ini
[DEFAULT]
ask = 
scan = true
```

---

The issue now - how to explain it - I have no idea. Most of the times, when I try to understand documentation for configurations online, I look at examples, as sometimes my English can't understand the author's mindset.
So all hope on adding documentation for the process I explained above is on you - I just can't think of a way to formulate it sadly in English (and I tried to write it in Hebrew and use translator - this was horrible lol).

Reply:
Personally I think it's fine as long as you give examples like you did, especially if you make the change I suggested above for the --ask option setup.

However, handling for hyphenated long options is slightly different (hyphens get translated to underscores iirc) but I'm not sure if any of the hyphenated options will likely be set in a config file at this point.

Reply:
I have tried to add some explanation to the `commit.all` example, hope it might give the needed pointer to the reader.